File: leetcode/solutions/1417.ReformatTheString/reformat_test.go
package solution

import (
	"testing"
)

func Test_1417_ReformatTheString(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  string
		out string
	}{
		{"a0b1c2", "a0b1c2"},
		{"leetcode", ""},
		{"1229857369", ""},
		{"covid2019", "c2o0v1i9d"},
		{"ab123", "1a2b3"},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := reformat(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/1417.ReformatTheString/reformat_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import reformat from "./reformat.ts";

Deno.test("1417. Reformat The String", () => {
  type test = [string, string];
  const tests: test[] = [
    ["a0b1c2", "a0b1c2"],
    ["leetcode", ""],
    ["1229857369", ""],
    ["covid2019", "c2o0v1i9d"],
    ["ab123", "1a2b3"],
  ];

  for (const [arg, want] of tests) {
    assertEquals(reformat(arg), want);
  }
});
